src/api/helpers/filter-ordnance-survey-addresses/index.test.ts

```diff
@@ -0,0 +1,72 @@
+import filterOrdnanceSurveyAddresses from '.';
+import mockOrdnanceSurveyResponse from '../../test-mocks/mock-ordnance-survey-response';
+
+const ordnanceSurveyResponse = mockOrdnanceSurveyResponse.results;
+
+describe('api/helpers/filter-ordnance-survey-addresses', () => {
+  describe('when an address has a SUB_BUILDING_NAME with a matching substring', () => {
+    it('should return an array with address', () => {
+      const [address] = ordnanceSurveyResponse;
+
+      const substring = String(address.DPA.SUB_BUILDING_NAME)?.substring(0, 5);
+
+      const result = filterOrdnanceSurveyAddresses(ordnanceSurveyResponse, substring);
+
+      const expected = [address];
+
+      expect(result).toEqual(expected);
+    });
+  });
+
+  describe('when an address has a BUILDING_NAME with a matching substring', () => {
+    it('should return an array with the address', () => {
+      const { 1: address } = ordnanceSurveyResponse;
+
+      const substring = String(address.DPA.BUILDING_NAME)?.substring(0, 5);
+
+      const result = filterOrdnanceSurveyAddresses(ordnanceSurveyResponse, substring);
+
+      const expected = [address];
+
+      expect(result).toEqual(expected);
+    });
+  });
+
+  describe('when an address has a ORGANISATION_NAME with a matching substring', () => {
+    it('should return an array with the address', () => {
+      const { 2: address } = ordnanceSurveyResponse;
+
+      const substring = String(address.DPA.ORGANISATION_NAME)?.substring(0, 5);
+
+      const result = filterOrdnanceSurveyAddresses(ordnanceSurveyResponse, substring);
+
+      const expected = [address];
+
+      expect(result).toEqual(expected);
+    });
+  });
+
+  describe('when an address has a BUILDING_NUMBER with a matching substring', () => {
+    it('should return an array with the address', () => {
+      const { 3: address } = ordnanceSurveyResponse;
+
+      const substring = String(address.DPA.BUILDING_NUMBER)?.substring(0, 5);
+
+      const result = filterOrdnanceSurveyAddresses(ordnanceSurveyResponse, substring);
+
+      const expected = [address];
+
+      expect(result).toEqual(expected);
+    });
+  });
+
+  describe('when no address is found', () => {
+    it('should return an empty array', () => {
```

Comment:
Can we add an assertion where `null` is returned from the function please.

Reply:
If `.filter` returns `null`, the `filtered` const returns an empty ar
ray so we cannot assert null values. I have double checked, it's handled in this unit test.[](url)

Test coverage for this is reported at 100%.

Reply:
The only reason I ask is you are returning null at LOC:28, ideally a test case which can reach that point in code.

Reply:
I think it's slightly overkill because `.filter` handles it natively, but I have split up individual address filtering into a new function `filterOrdnanceSurveyAddress`, so we can test the `null` return.